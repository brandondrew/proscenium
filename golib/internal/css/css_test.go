package css

import (
	"os"
	"testing"

	. "github.com/MakeNowJust/heredoc/dot"
	"github.com/gkampitakis/go-snaps/snaps"
)

func TestMain(t *testing.M) {
	v := t.Run()
	snaps.Clean(t)
	os.Exit(v)
}

func TestParseCss(t *testing.T) {
	parse := func(input string, path string) string {
		output, _ := ParseCss(D(input), path)
		t.Logf(output)
		return output
	}

	t.Run("should pass through regular css", func(t *testing.T) {
		snaps.MatchSnapshot(t, parse("body{}", "/foo.css"))
	})

	t.Run("css modules", func(t *testing.T) {
		t.Run("path is not a css module", func(t *testing.T) {
			snaps.MatchSnapshot(t, parse(`
				.title {
					color: green;
				}
			`, "/foo.css"))
		})

		t.Run("should rename classes", func(t *testing.T) {
			snaps.MatchSnapshot(t, parse(`
				.title {
					color: green;
				}
			`, "/foo.module.css"))
		})

		t.Run("should rename nested classes", func(t *testing.T) {
			snaps.MatchSnapshot(t, parse(`
				.title {
					color: green;
					.subtitle {
						color: blue;
					}
				}
			`, "/foo.module.css"))
		})

		t.Run("should rename compound classes", func(t *testing.T) {
			snaps.MatchSnapshot(t, parse(`
				.title.subtitle {
					color: green;
				}
			`, "/foo.module.css"))
		})

		t.Run("global function", func(t *testing.T) {
			t.Run("should not rename argument", func(t *testing.T) {
				snaps.MatchSnapshot(t, parse(`
					.title {
						color: blue;
					}
					:global(.subtitle) {
						color: green;
					}
					.author {
						color: red;
					}
				`, "/foo.module.css"))
			})

			t.Run("nested classes should be renamed", func(t *testing.T) {
				snaps.MatchSnapshot(t, parse(`
					:global(.subtitle) {
						color: green;
						.foo {
							color: orange;
						}
					}
				`, "/foo.module.css"))
			})
		})

		t.Run("global shorthand with argument", func(t *testing.T) {
			t.Run("should not rename argument", func(t *testing.T) {
				snaps.MatchSnapshot(t, parse(`
					.title {
						color: blue;
					}
					:global .subtitle {
						color: green;
					}
					.author {
						color: red;
					}
				`, "/foo.module.css"))
			})

			t.Run("nested classes should be renamed", func(t *testing.T) {
				snaps.MatchSnapshot(t, parse(`
					:global .subtitle {
						color: green;
						.foo {
							color: orange;
						}
					}
				`, "/foo.module.css"))
			})
		})

		t.Run("global shorthand without argument", func(t *testing.T) {
			t.Run("should rename all children", func(t *testing.T) {
				snaps.MatchSnapshot(t, parse(`
					:global {
						.subtitle {
							color: green;

							.foo {
								color: orange;
							}
						}
						.bar {
							color: blue;
						}
					}
					.author {
						color: red;
					}
				`, "/foo.module.css"))
			})
		})
	})

}
