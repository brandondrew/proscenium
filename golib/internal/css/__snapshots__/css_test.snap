
[TestParseCss/css_modules/path_is_not_a_css_module - 1]
.title {
    color: green;
}

---

[TestParseCss/css_modules/should_rename_classes - 1]
.title43c30152 {
    color: green;
}

---

[TestParseCss/css_modules/should_rename_nested_classes - 1]
.title43c30152 {
    color: green;
    .subtitle43c30152 {
        color: blue;
    }
}

---

[TestParseCss/css_modules/should_rename_compound_classes - 1]
.title43c30152.subtitle43c30152 {
    color: green;
}

---

[TestParseCss/css_modules/global_function/should_not_rename_function_argument - 1]
.title43c30152 {
    color: blue;
}
.subtitle {
    color: green;
}
.author43c30152 {
    color: red;
}

---

[TestParseCss/css_modules/global_function/nesting_in_function_body_should_be_renamed - 1]
.subtitle {
    color: green;
    .foo43c30152 {
        color: orange;
    }
}

---

[TestParseCss/css_modules/global_function/shorthand_with_argument/body_should_be_renamed - 1]
 .subtitle {
    color: green;
    .foo {
        color: orange;
    
}

---

[TestParseCss/css_modules/global_function/should_not_rename_argument - 1]
.title43c30152 {
    color: blue;
}
.subtitle {
    color: green;
}
.author43c30152 {
    color: red;
}

---

[TestParseCss/css_modules/global_function/nested_classes_should_be_renamed - 1]
.subtitle {
    color: green;
    .foo43c30152 {
        color: orange;
    }
}

---

[TestParseCss/css_modules/global_shorthand_with_argument/should_not_rename_argument - 1]
.title43c30152 {
    color: blue;
}
 .subtitle 
    color: green;
}
.author43c30152 {
    color: red;
}

---

[TestParseCss/css_modules/global_shorthand_with_argument/nested_classes_should_be_renamed - 1]
 .subtitle 
    color: green;
    .foo43c30152 {
        color: orange;
    }
}

---

[TestParseCss/css_modules/global_shorthand_without_argument/should_rename_all_children - 1]
 
    .subtitle {
        color: green;

        .foo {
            color: orange;
        }
    }
    .bar {
        color: blue;
    }
}
.author43c30152 {
    color: red;
}

---
